discovery_engine/lib/src/domain/models/document.dart

```diff
@@ -71,13 +74,27 @@ class Document {
 /// essentially if the user "liked" or "disliked" the document.
 @HiveType(typeId: documentFeedbackTypeId)
 enum DocumentFeedback {
-  @JsonValue(0)
-  @HiveField(0)
+  @JsonValue(RustUserReaction.Neutral)
+  @HiveField(RustUserReaction.Neutral)
   neutral,
-  @JsonValue(1)
-  @HiveField(1)
+  @JsonValue(RustUserReaction.Positive)
+  @HiveField(RustUserReaction.Positive)
   positive,
-  @JsonValue(2)
-  @HiveField(2)
+  @JsonValue(RustUserReaction.Negative)
+  @HiveField(RustUserReaction.Negative)
   negative,
 }
+
+extension DocumentFeedbackIntConversion on DocumentFeedback {
+  int toIntRepr() => _$DocumentFeedbackEnumMap[this]!;
+  static DocumentFeedback fromIntRepr(int intRepr) =>
+      $enumDecode(_$DocumentFeedbackEnumMap, intRepr);
+}
+
+// Without this `_$DocumentFeedbackEnumMap` is not generated.
```

Comment:
Can we use `@JsonEnum(alwaysCreate=true)` on the enum instead?

Reply:
yes, nice that they added that :+1: